spec/game_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'game'
+
+describe Game do
+  let(:game) { double :player_1 => "Jim", :player_weapon => :rock, :computer_weapon => :paper }
+  let(:weapon) { double :weapon }
+
+
+  describe '#player_1' do
+    it 'initializes with player name' do
+      expect(game.player_1).to eq "Jim"
+    end
+  end
+
+  describe '#player_weapon' do
+    it 'stores the players weapon choice' do
+      expect(game.player_weapon).to eq :rock
+    end
+  end
+
+  describe '#computer_weapon' do
+    it 'stores the computer weapon choice' do
+      expect(game.computer_weapon).to eq :paper
+    end
+  end
+
+  describe '#result' do
+    it 'displays the result of the game' do
+      allow(weapon).to receive(:result).with(:rock, :paper) {:Lose}
```

Comment:
The reason this test doesn't work is because, although `weapon` is mocked correctly, the method `game.result` still uses the weapon instance that was provided when instantiating `Game` at line 4

Looking at it now... it seems there's also a mismatch between how you create the `game` inside the app.rb file, and how you create it in this spec file (in one place, instances of `Weapon` are passed, in the other, symbol values). If we were to keep the `Weapon` class, we'd probably need to pass a `Weapon` instance to `Game.new` here, and this `weapon` would be the one mocked with the `allow(...)...` syntax